dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,11 +92,14 @@ public void join(WorkerInfo workerInfo) throws IOException {
     byte[] serializedEntity = entity.serialize();
     // If there's existing entry, check if it's me.
     if (existingEntityBytes != null) {
-      // It's not me, something is wrong.
+      // It's not me, or not the same me.
       if (!Arrays.equals(existingEntityBytes, serializedEntity)) {
-        // Might be regression of different formatted value of workerinfo is registered.
-        throw new AlreadyExistsException(
-            "Some other member with same id registered on the ring, bail.");
+        // In k8s this might be bcos worker pod restarting with the same worker identity
+        // but certain fields such as hostname has been changed. Register to ring path anyway.
+        LOG.warn("Same worker entity found bearing same workerid,"
+            + "maybe benign if pod restart in k8s env or same worker"
+            + " scheduled to restart on another machine in baremetal env.");
+        mAlluxioEtcdClient.createForPath(pathOnRing, Optional.of(serializedEntity));
```

Comment:
if you now observe a new worker with the same identity, would there be an old worker with the same id on the ring? If so then that obsolete one needs to be removed? Hmmmm but what about the corner case that the admin incorrectly used the same ID on multiple workers, like copying the config blindly?

Reply:
yes it will override,  for enterprise customer normally we should always do automated scripts such as ansible to run deployment instead of letting customer do stuff like copy properties. typical adding-node or new-deployment should always do a wipe-out clean to clean system info files such as worker_identity file

Reply:
nit: log the old one so we could tell from the logs if something is unexpected

Reply:
done